hoomd/hpmc/shape_move.py

```diff
@@ -0,0 +1,273 @@
+        enabled to allow the use of spherical depletants with shape moves.
+
+    Note:
+        Parameters must be given for every particle type and must be between 0
+        and 1. This class is limited to performing MC moves on the predefined
+        shape parameters, it does not performs any consistency checks
+        internally. Therefore, any shape constraint (e.g. constant volume, etc)
+        must be performed within the callback.
+
+    Example::
+
+        mc = hoomd.hpmc.integrate.ConvexPolyhedron()
+        mc.shape["A"] = dict(vertices=[(1, 1, 1), (-1, -1, 1),
+                                       (1, -1, -1), (-1, 1, -1)])
+        # example callback
+        class ExampleCallback:
+            def __init__(self):
+                default_dict = dict(sweep_radius=0, ignore_statistics=True)
+            def __call__(self, type, param_list):
+                # do something with params and define verts
+                return dict("vertices":verts, **self.default_dict))
+        move = hpmc.shape_move.ShapeSpace(callback = ExampleCallback)
+
+    Attributes:
+        callback (``callable`` [`str`, `list`], `dict` ]): The python function
+            that will be called to map the given shape parameters to a shape
+            definition. The function takes the particle type and a list of
+            parameters as arguments and return a dictionary with the shape
+            definition whose keys **must** match the shape definition of the
+            integrator:
+
+                ``callable[[str, list], dict]``
+
+        params (`TypeParameter` [``particle type``, `list`]): List of tunable
+            parameters to be updated. The length of the list defines the
+            dimension of the shape space for each particle type.
+
+        move_probability (`float`, optional): Average fraction of shape
+            parameters to change each timestep (**default**: 1).
+    """
+
+    _suported_shapes = {
+        'ConvexPolyhedron', 'ConvexSpheropolyhedron', 'Ellipsoid'
+    }
+
+    def __init__(self, callback, move_probability=1):
+
+        super().__init__(move_probability)
+        param_dict = ParameterDict(callback=object)
+        param_dict["callback"] = callback
+        self._param_dict.update(param_dict)
+
+        typeparam_shapeparams = TypeParameter('params',
+                                              type_kind='particle_types',
+                                              param_dict=TypeParameterDict(
+                                                  [float], len_keys=1))
+        self._add_typeparam(typeparam_shapeparams)
+
+
+class Vertex(ShapeMove):
+    """Apply shape moves where particle vertices are translated.
+
+    Args:
+        move_probability (`float`, optional): Average fraction of
+            vertices to change during each shape move (**default**: 1).
+
+    .. rubric:: Shape support.
+
+    The following shapes are supported:
+        * `hoomd.hpmc.integrate.ConvexPolyhedron`
```

Comment:
```suggestion
    The following shapes are supported:
    
    * `hoomd.hpmc.integrate.ConvexPolyhedron`
```
Not a definition list.